### PR TITLE
Use ubuntu-22.04 as target for Debian 12 with acfl and armpl-gcc inst…

### PR DESCRIPTION
**Pull request by dslarm**
@paolotricerri 
@annop-w 

On Debian12, acfl and armpl-gcc fall back to installing rhel-7 binaries, which doesn't work - this patch makes them use ubuntu-22.04 binaries which do work.

### CODE REVIEW

File: var/spack/repos/builtin/packages/armpl-gcc/package.py
Comment (paolotricerri) [inline var/spack/repos/builtin/packages/armpl-gcc/package.py:1]: For Debian12, we want to install a *.deb package and, currently, the default is *.rpm. I would add an entry in the _os_pkg_map dictionary mapping debian12 to deb package files.

**Review by paolotricerri — CHANGES_REQUESTED** [review]
I would add an entry into the _os_pkg_map that maps from debian12 to a *.deb package file.

File: var/spack/repos/builtin/packages/armpl-gcc/package.py
Comment (dslarm) [reply var/spack/repos/builtin/packages/armpl-gcc/package.py:1]: My bad - I missed that. Corrected and have pushed another commit to the branch.

**Comment by paolotricerri** [comment]
Looks good to me but I don't think I can merge. 

Maybe @alalazo can help?

Thanks

**Review by alalazo — APPROVED** [review]

**Pull request merged**